hazelcast/src/main/java/com/hazelcast/jet/JetService.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet;
+
+import com.hazelcast.core.HazelcastInstance;
+
+import com.hazelcast.jet.config.JetConfig;
+import com.hazelcast.jet.config.JobConfig;
+import com.hazelcast.jet.core.DAG;
+import com.hazelcast.jet.impl.AbstractJetInstance;
+import com.hazelcast.jet.impl.JobRepository;
+import com.hazelcast.jet.impl.SnapshotValidationRecord;
+import com.hazelcast.jet.pipeline.Pipeline;
+import com.hazelcast.map.IMap;
+import com.hazelcast.map.impl.MapService;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.List;
+
+import static com.hazelcast.jet.impl.JobRepository.exportedSnapshotMapName;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * Represents either an instance of a
+ * instance that connects to a remote cluster.
+ * @since 5.0
+ */
+public interface JetService {
+    /**
+     * Returns the underlying Hazelcast IMDG instance used by Jet. It will
+     * be either a server node or a client, depending on the type of this
+     * {@code JetInstance}.
+     */
+    @Nonnull
+    HazelcastInstance getHazelcastInstance();
```

Comment:
Do we need this method here? We don't have backlinks from other services, e.g. from `SqlService` or from `CPSubsystem`.

Reply:
I removed this backlink.